scripts/update_techniques.js

```diff
@@ -0,0 +1,142 @@
+      m.description = r.getCell(3).value;
+      m.url = r.getCell(4).value.hyperlink;
+      m.created_date = new Date(r.getCell(5).value);
+      m.updated_date = new Date(r.getCell(6).value);
+      m.version = Number(r.getCell(7).value);
+
+      mitigations.push(m);
+    }
+  });
+  console.log("Parsing relationships... ");
+  // read from relationships tab to assign mitigations to techniques
+  const relationshipSheet = wb.getWorksheet("relationships");
+  relationshipSheet.eachRow((r) => {
+    if (
+      r.number > 1 &&
+      r.getCell(1).value &&
+      r.getCell(1).value.charAt(0) === "M"
+    ) {
+      const mitigation = mitigations.find((m) => m.mid === r.getCell(1).value);
+      if (r.getCell(5).value.includes(".")) {
+        const subtechnique = subtechniques.find(
+          (t) => t.tid === r.getCell(5).value
+        );
+        subtechnique.mitigations.push(mitigation);
+      } else {
+        const technique = techniques.find((t) => t.tid === r.getCell(5).value);
+        technique.mitigations.push(mitigation);
+      }
+    }
+  });
+  console.log("Parsed relationships");
+  // add subtechniques to techniques
+  addSubtechniques(techniques, subtechniques);
+  console.log("Parsed subtechniques");
+  // read from the methodology tab to add scores to the technique objects
+  const scoreList = wb.getWorksheet("Methodology");
+  scoreList.eachRow((r) => {
+    const id = r.getCell(3).value;
+    if (id) {
+      const technique = techniques.find((t) => t.tid == id);
+      if (technique && technique.tid) {
+        technique.cumulative_score = r.getCell("B").value.result;
+        technique.adjusted_score = technique.cumulative_score;
+        technique.has_car = !!r.getCell("N").value;
+        technique.has_sigma = !!r.getCell("O").value;
+        technique.has_es_siem = !!r.getCell("P").value;
+        technique.has_splunk = !!r.getCell("Q").value;
+
+        technique.cis_controls = r.getCell("R").value
+          ? r.getCell("R").value.toString().split(",")
+          : [];
+        technique.nist_controls = r.getCell("T").value
+          ? r.getCell("T").value.toString().split(",")
+          : [];
+
+        technique.process_coverage = !!parseInt(r.getCell(31).value.result);
+        technique.network_coverage = !!parseInt(r.getCell(33).value.result);
+        technique.file_coverage = !!parseInt(r.getCell(35).value.result);
+        technique.cloud_coverage = !!parseInt(r.getCell(37).value.result);
+        technique.hardware_coverage = !!parseInt(r.getCell(39).value.result);
+      }
+    }
+  });
+  console.log("Parsed scores from Methodology page");
+  // export technique list to JSON
+  const str = JSON.stringify(techniques, null, 4);
+  fs.writeFile(DESTINATION_FILE, str, (error) => {
+    if (error) {
+      console.error(error);
+      throw error;
+    }
+  });
+  console.log("Export technique data to Techniques.json");
+});
+
+function addSubtechniques(techniques, subtechniques) {
```

Comment:
This function is only invoked once, so I'm not sure I see the point of breaking this logic out (into a separate function) over inlining it with the rest of your code above. I would either: 
 1. Extract the logic in this function and inline it with the rest of your code above.
 2. Breakout each step of this process into separate functions.
    - (e.g. `parseTechniques()`, `parseMitigations()`, `parseRelationships()` etc.)